src/EnergyPlus/ScheduleManager.cc

```diff
@@ -4491,6 +4491,78 @@ namespace ScheduleManager {
+        constexpr std::array<bool, maxDayTypes> dayTypeFilterWeHo = {true, false, false, false, false, false, true, true, false, false, false, false};
+        //  Sun    Mon    Tues   Wed    Thur   Fri    Sat    Hol    Summer Winter Cust1  Cust2
+        constexpr std::array<bool, maxDayTypes> dayTypeFilterSumDsDy = {
+            false, false, false, false, false, false, false, false, true, false, false, false};
+        constexpr std::array<bool, maxDayTypes> dayTypeFilterWinDsDy = {
+            false, false, false, false, false, false, false, false, false, true, false, false};
+        constexpr std::array<bool, maxDayTypes> dayTypeFilterNone = {
+            false, false, false, false, false, false, false, false, false, false, false, false};
+        if (ScheduleIndex > 0 && ScheduleIndex <= state.dataScheduleMgr->NumSchedules) {
+            int curDayTypeGroup = static_cast<int>(days);
+            auto &curSch = state.dataScheduleMgr->Schedule(ScheduleIndex);
+            if (!curSch.MaxMinByDayTypeSet[curDayTypeGroup]) {
+                std::array<bool, maxDayTypes> dayTypeFilter;
+                switch (days) {
+                case DayTypeGroup::Weekday:
+                    dayTypeFilter = dayTypeFilterWkDy;
+                    break;
+                case DayTypeGroup::WeekEndHoliday:
+                    dayTypeFilter = dayTypeFilterWeHo;
+                    break;
+                case DayTypeGroup::SummerDesignDay:
+                    dayTypeFilter = dayTypeFilterSumDsDy;
+                    break;
+                case DayTypeGroup::WinterDesignDay:
+                    dayTypeFilter = dayTypeFilterWinDsDy;
+                    break;
+                default:
+                    dayTypeFilter = dayTypeFilterNone;
+                    break;
+                }
+                for (int iDayOfYear = 1; iDayOfYear <= 366; ++iDayOfYear) {
+                    int WkSch = curSch.WeekSchedulePointer(iDayOfYear);
+                    auto &weekSch = state.dataScheduleMgr->WeekSchedule(WkSch);
+                    for (int jType = 1; jType <= maxDayTypes; ++jType) {
+                        if (dayTypeFilter[jType - 1]) {
+                            auto &daySch = state.dataScheduleMgr->DaySchedule(weekSch.DaySchedulePointer(jType));
+                            MinValue = min(MinValue, minval(daySch.TSValue));
+                            MaxValue = max(MaxValue, maxval(daySch.TSValue));
+                        }
+                    }
+                }
+                if (MinValue == Constant::BigNumber) MinValue = 0;
+                if (MaxValue == -Constant::BigNumber) MaxValue = 0;
+                // store for the next call of the same schedule
+                curSch.MaxByDayType[curDayTypeGroup] = MaxValue;
+                curSch.MinByDayType[curDayTypeGroup] = MinValue;
+                curSch.MaxMinByDayTypeSet[curDayTypeGroup] = true;
```

Comment:
There is another loop over people schedules in DetermineSystemPopulationDiversity. Not sure if this could help there or if when checking those people schedules these 3 values could be set to save even more simulation time.